rmf_site_editor/src/interaction/cursor.rs

```diff
@@ -120,6 +120,26 @@ impl Cursor {
         }
     }
 
+    // TODO(luca) reduce duplication here
+    pub fn set_model_instance_preview(
+        &mut self,
+        commands: &mut Commands,
+        model_loader: &mut ModelLoader,
+        model_instance: Option<ModelInstance<Entity>>,
+    ) {
+        self.remove_preview(commands);
+        self.preview_model = if let Some(model_instance) = model_instance {
+            Some(
+                model_loader
+                    .spawn_model_instance(self.frame, model_instance.clone())
+                    .insert(Pending)
+                    .id(),
+            )
+        } else {
+            None
+        };
+    }
+
     pub fn set_model_preview(
         &mut self,
         commands: &mut Commands,
```

Comment:
[f527790](https://github.com/open-rmf/rmf_site/pull/242/commits/f52779060abdfd2a4162bbc539d1f6c1d3212b09), I removed the now-unused `spawn_model` and `spawn_model_impulse` as well, let me know if that should be reverted for any reason

Reply:
I think that's fair, the downstream workcell editor will probably need to be rethought a bit and it's unclear whether these function even work right now since they are not used anywhere.